app/models/post.rb

```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'post_id'
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+
+  def posts_counter(user)
+    array = user.posts
+    user.posts_counter = array.length
+  end
```

Comment:
**OPTIONAL**
Please use the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#posts_counter` method every time a new post has been saved to the database, then make the `posts_counter` method private.